saleor/plugins/manager.py

```diff
@@ -95,19 +95,21 @@ def __run_method_on_single_plugin(
         will return previous_value.
         """
         plugin_id = getattr(plugin, "PLUGIN_ID", "")
+        plugin_method = getattr(plugin, method_name, NotImplemented)
+        if plugin_method == NotImplemented:
+            return previous_value
         with opentracing.global_tracer().start_active_span(
-            f"{plugin_id}.{method_name}"
-        ):
-            plugin_method = getattr(plugin, method_name, NotImplemented)
-            if plugin_method == NotImplemented:
-                return previous_value
-
+            f"{plugin_id}.{method_name}", finish_on_close=False
+        ) as scope:
             returned_value = plugin_method(
                 *args, **kwargs, previous_value=previous_value
             )
             if returned_value == NotImplemented:
                 return previous_value
-            return returned_value
+            # We should create spans for plugin_method with return data instead of
+            # `NotImplemented`
+            scope.span.finish()
```

Comment:
So if `NotImplemented` is returned, we create and report an unfinished span. I believe that's undefined behavior at best.